cql3/statements/modification_statement.cc

```diff
@@ -439,13 +441,30 @@ modification_statement::process_where_clause(data_dictionary::database db, expr:
     }
 }
 
+::shared_ptr<strongly_consistent_modification_statement>
+modification_statement::prepare_for_raft() const {
+    throw raft::broadcast_tables::unsupported_operation_error{};
+}
+
 namespace raw {
 
+::shared_ptr<cql_statement_opt_metadata>
+modification_statement::prepare_statement(data_dictionary::database db, prepare_context& ctx, cql_stats& stats) {
```

Comment:
The number of different functions named `prepare_*` make my head hurt.

I would prefer this to be inlined in `prepare`, don't see much advantage to introducing yet another helper function (which is very simple and used in exactly one place).

But I'll leave this decision to you.

Reply:
`modifiation_statement` has to be prepared and then it prepares `strongly_consistent_modifiation_statement`, which doesn't derive from `modifiation_statement`. Thus we can't reassign the `statement` variable (like you mentioned before) and we can't change variable's type (we use the `prepare_for_raft` method). I would leave it in a separate method.

Reply:
Ah, so we would need two variables to inline it.
Meh, ok, separate method is fine.